src/scenes/Dashboard/ProgressReportsWidget/ProgressReportsExpandedGrid.tsx

```diff
@@ -128,6 +138,21 @@ export const ProgressReportsExpandedGrid = (
     [onMouseDown]
   );
 
+  const onStateChange = useDebounceFn(
+    () => {
+      const gridState = apiRef.current.exportState();
+      if (!isEqual(gridState, prevState.current)) {
+        prevState.current = gridState;
+        setSavedGridState(gridState);
+      }
```

Comment:
`ahooks` has a `usePrevious` hook, that can accept a `shouldUpdate` where `isEqual` can be passed.
Maybe that can simplify most of this?

I guess you'd still need an effect to push when there is an actual change.

Why actually bother with this? It's to prevent a re-render with the state change of session storage?

Reply:
I can look into it and see about cleaning this up.